### PR TITLE
Higher order function workshop

### PR DESCRIPTION
**Pull request by AlaaTaima**

### CODE REVIEW

File: coursebook/higherOrderFunctions.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+> ### Try another methods alone 👨‍💻
+<hr>
+
+### 🔸  Creating Our own Higher-Order Function :
+<br>
+
+Up until this point, we saw various Higher-order functions built into the language.
+
+ Now let’s create our own Higher-order function.
+ <br><br>
```
Comment (yosefanajjar) [inline coursebook/higherOrderFunctions.md:56]:
```suggestion
 <br />
```

File: coursebook/higherOrderFunctions.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+Look at the code below,👇 the `filter()` take a function as an argument.
+
+```javascript
+var ages = [32, 33, 12, 40];
+
+function checkAges(age) {
+  return age >= 30
+}
+
+var newarray = ages.filter(checkAges);
```
Comment (yosefanajjar) [inline coursebook/higherOrderFunctions.md:44]:
```suggestion
var newArray = ages.filter(checkAges);
```

File: coursebook/higherOrderFunctions.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+### 🔹 map() method : 
+
+Look at the code below,👇 the `map()` take a function as an argument.
+
+```javascript
+var numbers = [65, 44, 12, 4];
+function myFunction(num) {
+  return num * 10;
+}
+var newarray = numbers.map(myFunction);
```
Comment (yosefanajjar) [inline coursebook/higherOrderFunctions.md:30]:
```suggestion
var newArray = numbers.map(myFunction);
```

**Review by yosefanajjar — DISMISSED** [review]
Great job! just left some suggestions, it would be nice if you use the gifs from this article https://jst.hashnode.dev/javascript-higher-order-functions-map-filter-and-reduce about higher-order functions and reference to it for copy rights reasons

**Comment by AlaaTaima** [comment]
> Great job! just left some suggestions, it would be nice if you use the gifs from this article https://jst.hashnode.dev/javascript-higher-order-functions-map-filter-and-reduce about higher-order functions and reference to it for copy rights reasons

Thanks, I will see them.

**Review by sara219 — APPROVED** [review]

**Review by hshahwan — DISMISSED** [review]

**Review by NoureldeanSaed — APPROVED** [review]

**Review by sara219 — APPROVED** [review]

**Review by NoureldeanSaed — APPROVED** [review]

**Pull request merged**